### PR TITLE
add hostport to apisix

### PR DESCRIPTION
**Pull request by SRCchen**

### CODE REVIEW

File: charts/apisix/templates/deployment.yaml [hunk 1]
```diff
@@ -65,9 +65,11 @@ spec:
             - name: http
               containerPort: {{ .Values.gateway.http.containerPort }}
               protocol: TCP
+              hostPort: {{ .Values.gateway.http.hostPort }}
```
Comment (tokers) [inline charts/apisix/templates/deployment.yaml:68]: Here we should add an `if` block to check whether the `hostPort` is configured. Or hostPort is always enabled.

File: charts/apisix/values.yaml [hunk 2]
```diff
@@ -135,10 +135,12 @@ gateway:
     enabled: true
     servicePort: 80
     containerPort: 9080
+    hostPort: 80
```
Comment (tokers) [inline charts/apisix/values.yaml:138]: This field should be commented by default.

**Comment by SRCchen** [comment]
`hostport` have been disabled by default   @tokers

**Review by tokers — COMMENTED** [review]
LGTM except we need to add some descriptions in https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix/README.md.

File: charts/apisix/README.md [hunk 3]
```diff
@@ -89,9 +89,11 @@ Apache APISIX service parameters, this determines how users can access itself.
 | `gateway.type`                  | Apache APISIX service type for user access itself                                                                                                                                   | `NodePort` |
 | `gateway.externalTrafficPolicy` | Setting how the Service route external traffic                                                                                                                                      | `Cluster`  |
 | `gateway.http`                  | Apache APISIX service settings for http                                                                                                                                             |            |
+| `gateway.http.hostPort`         | Set hostPort to access Apache APISIX service for http                                                                                                                               | `Not Set`  |
```
Comment (tokers) [inline charts/apisix/README.md:92]: If we don't set a default value, just leave it empty.

Comment (tokers) [inline charts/apisix/README.md:92]:
```suggestion
| `gateway.http.hostPort`         | Specify a host port and expose the HTTP port of Apache APISIX to it |   |
```

File: charts/apisix/README.md [hunk 4]
```diff
@@ -89,9 +89,11 @@ Apache APISIX service parameters, this determines how users can access itself.
...
 | `gateway.tls`                   | Apache APISIX service settings for tls                                                                                                                                              |            |
 | `gateway.tls.existingCASecret`  | Specifies the name of Secret contains trusted CA certificates in the PEM format used to verify the certificate when APISIX needs to do SSL/TLS handshaking with external services (e.g. etcd) | `""`       |
 | `gateway.tls.certCAFilename`    | filename be used in the `gateway.tls.existingCASecret`                                                                                                                                          | `""`       |
+| `gateway.tls.hostPort`          | Set hostPort to access Apache APISIX service for tls                                                                                                                                | `Not Set`  |
```
Comment (tokers) [inline charts/apisix/README.md:96]:
```suggestion
| `gateway.tls.hostPort`          | Specify a host port and expose the HTTPS port of Apache APISIX to it |   |
```

Comment (SRCchen) [reply charts/apisix/README.md]: thanks

**Review by tokers — APPROVED** [review]

**Pull request closed without merging**